File: C++ solutions/1518. Water Bottles.cpp
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int total = numBottles;

        while (numBottles >= numExchange) {
            total += numBottles / numExchange;
            numBottles = (numBottles / numExchange) + (numBottles % numExchange);
        }
        return total;
    }
};

File: Python solutions/1518. Water Bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        total = numBottles

        while numBottles >= numExchange :
            total += numBottles // numExchange
            numBottles = (numBottles // numExchange) + (numBottles % numExchange)

        return total